pages/post/[postId].tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { getShortUrl, getPostDate } from '../../utils/utils';
+
+interface PostProps {
+  postID: string;
+  author: {
+    userName: string;
+  };
+  title: string;
+  url: URL;
+  description: string;
+  upvotes: number;
+  timeCreated: number;
+}
+
+const ViewPost = () => {
+  const router = useRouter();
+  const { postId } = router.query;
+  const [post, setPost] = useState<PostProps>();
+
+  useEffect(() => {
+    const getPost = async () => {
+      try {
+        const response = await fetch('/api/getAllPosts');
+        const allData = await response.json();
+        const { posts } = allData.data;
+        setPost(
+          () => posts.filter((post: PostProps) => post.postID === postId)[0]
+        );
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    getPost();
+  }, [postId]);
+
+  return (
+    <>
+      {post ? (
+        <>
+          <div>
+            <p>upvote</p>
+            <p>{post.title}</p>
+            <p>({getShortUrl(new URL(post.url))})</p>
```

Comment:
Lol, you're doing an URL and then creating another URL inside `getShortUrl`

easily abstracted to
```
const getShortUrl = (url: string) => {
  const { hostname } = new URL(url);
  return hostname;
};
```

```getShortUrl(post.url)```